src/http/oauth2.ts

```diff
@@ -113,11 +124,24 @@ export function createOAuth2HttpClient(
   httpWithAuth.interceptors.response.use(transformResponseData);
   httpWithAuth.interceptors.response.use(transformKeysResponseData);
 
-  async function setTokenData(data: TokenDataOauth2) {
+  /**
+   * - Adds a creationTimestamp to the token
+   * - Notifies the user
+   * - Stores the token to authenticate future requests
+   * - Returns the token
+   */
+  async function dateAndSetTokenData(data: TokenDataOauth2) {
```

Comment:
Name can be discussed for sure, started off with `setAndHydrateTokenData`.
(But maybe the whole concept of adding the timestamp and returning is not great. See other comment)